server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java

```diff
@@ -101,11 +128,9 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       String rawUpdateMask = (String) mappedRequest.getOrDefault(UPDATE_MASK_FIELD_NAME, null);
 
       // Get the Person from the database and convert it to JSON for easy manipulation.
-      String userId = (String) personInfo.getOrDefault(Person.USER_ID_FIELD_NAME, null);
-      if (userId == null) {
-        throw new IllegalArgumentException(
-            String.format(NO_FIELD_ERROR, Person.USER_ID_FIELD_NAME));
-      }
+      String idToken = (String) mappedRequest.get(ID_TOKEN_FIELD_NAME);
+      String userId = AuthenticationHelper.getUserIdFromIdToken(idToken, verifier);
```

Comment:
Should we validate that the userId included in the request matches the userId in the idToken?

I don't think there's a risk of us updating the wrong person, but a mismatch should probably return an error.

Reply:
That's a very good point. I've added this (plus test for it) into the code.